tests/unit/test_logging.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file tests/test_logging.cpp
+ * @brief Test src/logging.*.
+ */
+#include <fstream>
+
+#include <src/logging.h>
+
+#include <tests/conftest.cpp>
+
+class LoggerInitTest: public virtual BaseTest, public ::testing::WithParamInterface<int> {
+protected:
+  void
+  SetUp() override {
+    BaseTest::SetUp();
+  }
+
+  void
+  TearDown() override {
+    BaseTest::TearDown();
+  }
+};
+INSTANTIATE_TEST_SUITE_P(
+  LogLevel,
+  LoggerInitTest,
+  ::testing::Values(
+    0,
+    1,
+    2,
+    3,
+    4,
+    5));
+TEST_P(LoggerInitTest, InitLogging) {
+  int logLevel = GetParam();
+  std::string logFilePath = "test_log_" + std::to_string(logLevel) + ".log";
+
+  auto log_deinit = logging::init(logLevel, logFilePath);
+  if (!log_deinit) {
+    FAIL() << "Failed to initialize logging";
+  }
+}
+
+TEST(LogFlushTest, CheckLogFile) {
+  // todo: why do we need to initialize logging here, when it's already done in the BaseTest fixture?
```

Comment:
TODO: because the `deinit_guard` goes out of scope and immediately cleans up as it should.